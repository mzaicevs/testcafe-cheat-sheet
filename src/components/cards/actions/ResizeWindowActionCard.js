import React from "react";
import { actions_resizeWindow } from "../../../snippets";
import CodeCard from "../../../base_components/CodeCard";

const ResizeWindowActionCard = () => (
  <CodeCard
    cardTitle={actions_resizeWindow.title}
    fullDocLink={actions_resizeWindow.fullDocLink}
    codeSnippet={{
      code: actions_resizeWindow.code,
      language: actions_resizeWindow.language,
      options: actions_resizeWindow.options,
      example: actions_resizeWindow.example
    }}
    description={actions_resizeWindow.description}
  />
);

export default ResizeWindowActionCard;
