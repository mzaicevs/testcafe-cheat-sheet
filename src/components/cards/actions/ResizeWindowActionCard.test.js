import React from "react";
import { shallow } from "enzyme";
import ResizeWindowActionCard from "./ResizeWindowActionCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("ResizeWindowActionCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<ResizeWindowActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
