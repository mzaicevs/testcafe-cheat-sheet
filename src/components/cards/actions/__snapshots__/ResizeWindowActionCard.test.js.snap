// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizeWindowActionCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Resize Window - Action"
  codeSnippet={
    Object {
      "code": Array [
        "t.resizeWindow( width, height )",
        "t.resizeWindowToFitDevice( deviceName [, options] )",
        "t.maximizeWindow( )",
      ],
      "example": "import { Selector } from 'testcafe';

  const menu = Selector('#side-menu');
    
  fixture \`My fixture\`
    .page \`http://www.example.com/\`;
    
  test('Side menu disappears on small screens', async t => {
    await t
      .resizeWindow(200, 100)
      .expect(menu.getStyleProperty('display')).eql('none');
  });
    
  test('Header is displayed on Xperia Z in portrait', async t => {
    await t
      .resizeWindowToFitDevice('Sony Xperia Z', {
        portraitOrientation: true
      })
      .expect(header.getStyleProperty('display')).notEql('none');
  });
    
  test('Side menu is displayed in full screen', async t => {
    await t
      .maximizeWindow()
      .expect(menu.visible).ok();
  });",
      "language": "javascript",
      "options": "{
    portraitOrientation: Boolean
    /** true for portrait - false for landscape
  }",
    }
  }
  description="Resize the window to a specific pixel dimension, or to fit on the screen of the specified mobile device, or just to maximise"
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/actions/resize-window.html"
/>
`;
